scheduler/src/cook/components.clj

```diff
@@ -294,10 +280,27 @@
                            (throw (ex-info "Framework id not configured and not in ZooKeeper" {})))
                          (log/info "Using framework id:" framework-id)
                          framework-id)))
-     :compute-clusters (fnk [settings]
-                         ((util/lazy-load-var 'cook.mesos.mesos-compute-cluster/setup-compute-cluster-map-from-config) datomic/conn settings))
+     :compute-clusters (fnk [exit-code-syncer-state
+                             mesos-heartbeat-chan
+                             sandbox-syncer-state
+                             settings
+                             trigger-chans]
+                         (let [constructor (util/lazy-load-var 'cook.mesos.mesos-compute-cluster/->MesosComputeCluster)
+                               create-mesos-compute-cluster (fn [compute-cluster-name framework-id db-id driver-atom]
+                                                              (constructor compute-cluster-name
+                                                                           framework-id
+                                                                           db-id
+                                                                           driver-atom
+                                                                           sandbox-syncer-state
+                                                                           exit-code-syncer-state
+                                                                           mesos-heartbeat-chan
+                                                                           trigger-chans))]
+                           ((util/lazy-load-var 'cook.mesos.mesos-compute-cluster/setup-compute-cluster-map-from-config) datomic/conn settings
+                             create-mesos-compute-cluster)))
      :mesos-datomic-mult (fnk []
                            (first ((util/lazy-load-var 'cook.datomic/create-tx-report-mult) datomic/conn)))
+     :mesos-heartbeat-chan (fnk []
```

Comment:
Can we add a todo that this mesos-specific stuff will be scoped down?

Reply:
Done